src/Moq/Mock.cs

```diff
@@ -813,7 +813,7 @@ internal void AddInnerMockSetup(MethodInfo method, IReadOnlyList<Expression> arg
 				var property = expression.ToPropertyInfo();
 				Guard.CanRead(property);
 
-				Debug.Assert(method == property.GetGetMethod(true));
+				Debug.Assert(property.CanRead(out var getter) && method == getter);
```

Comment:
It's hard to tell whether this change is correct in all situations&mdash;but it doesn't break any tests, so let's go with it for now.